### ISSUE TITLE
java.lang.IllegalStateException: Updater has not been initialized with view state

### ISSUE THREAD

**Issue by SRTjiawei** [deeplearning4j/deeplearning4j#2012]
`java.lang.IllegalStateException: Updater has not been initialized with view state
	at org.nd4j.linalg.learning.RmsProp.getGradient(RmsProp.java:56)
	at org.deeplearning4j.nn.updater.BaseUpdater.update(BaseUpdater.java:78)
	at org.deeplearning4j.nn.updater.graph.ComputationGraphUpdater.update(ComputationGraphUpdater.java:135)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.updateGradientAccordingToParams(BaseOptimizer.java:282)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:155)
	at org.deeplearning4j.optimize.solvers.StochasticGradientDescent.optimize(StochasticGradientDescent.java:54)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:51)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:769)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:737)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:612)`

I got this exception when calling fit() of the networks in a for loop:
`
for (DataSet ds : dataSets) {
                    net.fit(ds);
                }
`
Might be caused by a invalid networks configuration, but should it be picked up earlier?

**Comment by SRTjiawei** [comment]
The networks configuration is as follow (just a toy conf I tried):
`ComputationGraphConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .iterations(10)
                .learningRate(0.01)
                .regularization(true).l2(0.001)
                .weightInit(WeightInit.XAVIER)
                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                .updater(Updater.RMSPROP).rmsDecay(0.95)
                .graphBuilder()
                .addInputs("input") // can use any label for this
                .setInputTypes(
                        InputType.convolutional(103, dimention, 1))
                .addLayer("conv", new ConvolutionLayer.Builder(3, dimention)
                        .nIn(nChannels)
                        .stride(1, 1)
                        .nOut(2)
                        .activation("identity")
                        .build(), "input")
                .addLayer("pooling", new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
                        .stride(2, 1)
                        .build(), "conv")
                .addLayer("out", new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .nOut(outputNum)
                        .activation("softmax")
                        .build(), "pooling")
                .setOutputs("out")
                .pretrain(false)
                .backprop(true)
                .build();`
The DataSet I used is the same format as it is in the MnistDataSetIterator.

**Comment by AlexDBlack** [comment]
So this is (was) a bug in how updaters are handled with subsampling layers in ComputationGraph.
It was fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/1944

I tested with your config on 0.5.0 (bug present) and current dl4j master (definitely fixed).
